source/Calamari.Common/Plumbing/Variables/IVariables.cs

```diff
@@ -10,13 +10,17 @@ public interface IVariables : IEnumerable<KeyValuePair<string, string>>
         bool IsSet(string name);
         void Set(string name, string? value);
         void SetStrings(string variableName, IEnumerable<string> values, string separator);
-        string GetRaw(string variableName);
+        string? GetRaw(string variableName);
         [return: NotNullIfNotNull("defaultValue")]
         string? Get(string variableName, string? defaultValue = null);
         [return: NotNullIfNotNull("defaultValue")]
-        string? Get(string variableName, out string error, string? defaultValue = null);
-        string Evaluate(string expressionOrVariableOrText, out string error, bool haltOnError = true);
-        string Evaluate(string expressionOrVariableOrText);
+        string? Get(string variableName, out string? error, string? defaultValue = null);
+
+        [return: NotNullIfNotNull("expressionOrVariableOrText")]
```

Comment:
This sucks a bit, looks like .Net Core will improve this in the future https://github.com/dotnet/csharplang/issues/373

Reply:
yes, I did try `nameof`, just to be sure it didn't work :)